src/parser/error/input.rs

```diff
@@ -74,6 +74,24 @@ impl ErrorInput for ParserInput<'_> {
     }
 }
 
+impl ErrorInput for &Vec<TokenWithLocation> {
+    fn line(&self) -> u32 {
+        self.as_slice().line()
+    }
+
+    fn column(&self) -> usize {
+        self.as_slice().column()
+    }
+
+    fn snippet(&self) -> String {
+        self.as_slice().snippet()
+    }
+
+    fn is_empty(&self) -> bool {
+        self.as_slice().is_empty()
+    }
+}
+
```

Comment:
Minor nit, but this feels like avoidable duplication. Should double check that this works, but I think you can do this to delegate to the non-reference version:

```suggestion
impl ErrorInput for &Vec<TokenWithLocation> {
    fn line(&self) -> u32 {
        (*self).line()
    }

    fn column(&self) -> usize {
        (*self).column()
    }

    fn snippet(&self) -> String {
        (*self).snippet()
    }

    fn is_empty(&self) -> bool {
        (*self).is_empty()
    }
}

```

Reply:
I think this is just an artifact of trying to reconcile my changes with rebasing against your branch. There were quite a lot of conflicts I had to manually resolve. Looks like this one was done incorrectly.

I'll try to remove it and make sure that doesn't cause issues.